assets/scripts/dialogs/UpgradeDialog.jsx

```diff
@@ -1,56 +1,115 @@
 import React, { Component } from 'react'
+import axios from 'axios'
+import { injectIntl, intlShape, FormattedMessage } from 'react-intl'
+
 import StripeCheckout from 'react-stripe-checkout'
-// import { FormattedMessage } from 'react-intl'
+import PropTypes from 'prop-types'
 import Dialog from './Dialog'
-import { STRIPE_API_KEY } from '../app/config'
-// import './UpgradeDialog.scss'
+import { API_URL, STRIPE_API_KEY } from '../app/config'
+import { connect } from 'react-redux'
+import './UpgradeDialog.scss'
+
+const DEFAULT_BODY = 'Thank you for using Streetmix! For only $5/month, the Enthusiast Plan lets users support Streetmix while also gaining access to new experimental features. Plus your avatar gets a neat badge!'
 
 class UpgradeDialog extends Component {
   scriptContainerEl = React.createRef()
 
-  onToken = (token, addresses) => {
-    // TODO: Send the token information and any other
-    // relevant information to your payment process
-    // server, wait for the response, and update the UI
-    // accordingly. How this is done is up to you. Using
-    // XHR, fetch, or a GraphQL mutation is typical.
-    console.log('onToken', token, addresses)
+  constructor (props) {
```

Comment:
Can we refactor this component into a functional component?
- Then we can use the `useIntl` hook, and we don't need `injectIntl` or `intlShape` (see this example: https://github.com/streetmix/streetmix/pull/1490/files)
- We can use the `useState` hook for state
- We have no other lifecycle methods here, so a functional component can work

What do you think?